src/main/kotlin/graphics/scenery/controls/behaviours/MouseDragPlane.kt

```diff
@@ -0,0 +1,140 @@
+/*-
+ * #%L
+ * Scenery-backed 3D visualization package for ImageJ.
+ * %%
+ * Copyright (C) 2016 - 2020 SciView developers.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package graphics.scenery.controls.behaviours
+
+import graphics.scenery.BoundingGrid
+import graphics.scenery.Camera
+import graphics.scenery.Node
+import graphics.scenery.utils.LazyLogger
+import org.joml.Vector3f
+import org.scijava.ui.behaviour.DragBehaviour
+import org.scijava.ui.behaviour.ScrollBehaviour
+import kotlin.reflect.KProperty
+
+/**
+ * Drag nodes along the viewplane by mouse.
+ *
+ * @author Kyle Harrington
+ * @author Jan Tiemann
+ */
+open class MouseDragPlane(
+    protected val name: String,
+    protected val camera: () -> Camera?,
+    protected val alternativeTargetNode: (() -> Node?)? = null,
+    protected var debugRaycast: Boolean = false,
+    protected var ignoredObjects: List<Class<*>> = listOf<Class<*>>(BoundingGrid::class.java),
+    protected val mouseSpeed: () -> Float = { 0.25f },
+    protected val FPSSpeedSlow: () -> Float = { 0.05f }
+) : DragBehaviour, ScrollBehaviour {
+
+    protected val logger by LazyLogger()
+    protected val cam: Camera? by CameraDelegate()
+
+    protected var currentNode: Node? = null
+    private var lastX = 0
+    private var lastY = 0
+
+
+    /** Camera delegate class, converting lambdas to Cameras. */
+    protected inner class CameraDelegate {
```

Comment:
The CameraDelegate pops up in two classes, MouseDragPlane, and MouseDragSphere. Is there a way to maybe unify this?

Reply:
extracted camera delegate in all applicable classes